src/main/java/com/direwolf20/buildinggadgets/common/util/helpers/NBTHelper.java

```diff
@@ -345,8 +347,31 @@ public static NBTTagList createStringList(String[] strings) {
     }
 
     /**
-     * If the given stack has a tag, returns it. If the given stack does not have a tag, it will set a reference and return the new tag
-     * compound.
+     * Connect two {@link NBTTagList} together to create a new one. This process has no side effects, which means it
+     * will not modify two parameters.
+     * <p>
+     * Additionally, if any of the lists are empty, the method will directly return the nonempty one. If both of them are
+     * empty, it will directly return the second list.
+     */
+    public static NBTTagList concat(NBTTagList first, NBTTagList second) {
+        if (first.isEmpty())
```

Comment:
I'm all about dat clean code look

```suggestion
        if (first.isEmpty() || second.isEmpty())
               return first.isEmpty() ? second : first;
```

Reply:
@MichaelHillcox in my opinion your suggestion looks less straight forward

Reply:
I said clean not simple :P haha

Reply:
Is is logically weird, you are examining whether the lists are empty twice.

Reply:
Not really? I'm not going to reject the pr based on this. Mostly just a suggestion

Reply:
Personally I don't understand what is this (the suggestion) on a swipe through.

Reply:
I agree with hn...

Reply:
Well, logically speaking, the entire if statement is redundant. Concatenating 2 arrays with always produce the right result even if one is empty. It'll just combine one empty one and one not empty one.

Reply:
It matches specification though... If one is empty: return the other.
(Said specification makes sense, as it does not make sense to perform O(n) with n being the non-empty-length Operations just to return a copy (when we are only asked for some form of concat))

Reply:
Well then... Either way haha